src/optimagic/visualization/slice_plot.py

```diff
@@ -66,63 +75,148 @@ def slice_plot(
-            defined in the function are used.
-        batch_evaluator (str or callable): See :ref:`batch_evaluators`.
-
+        title: The figure title.
+        return_dict: If True, return dictionary with individual plots of each parameter,
+            else, combine individual plots into a figure with subplots.
+        make_subplot_kwargs: Dictionary of keyword arguments used to instantiate plotly
+            Figure with multiple subplots. Is used to define properties such as, for
+            example, the spacing between subplots (governed by 'horizontal_spacing' and
+            'vertical_spacing'). If None, default arguments defined in the function are
+            used.
+        batch_evaluator: See :ref:`batch_evaluators`.
 
     Returns:
-        out (dict or plotly.Figure): Returns either dictionary with individual slice
-            plots for each parameter or a plotly Figure combining the individual plots.
+        The figure object containing the slice plot if `return_dict` is False.
+            Otherwise, a dictionary with individual slice plots for each parameter.
 
     """
+    # ==================================================================================
+    # Process inputs
+
     bounds = replace_and_warn_about_deprecated_bounds(
         lower_bounds=lower_bounds,
         upper_bounds=upper_bounds,
         bounds=bounds,
     )
-
     bounds = pre_process_bounds(bounds)
 
-    layout_kwargs = None
-    if title is not None:
-        title_kwargs = {"text": title}
+    func, func_eval = _get_processed_func_and_func_eval(func, func_kwargs, params)
+
+    # ==================================================================================
+    # Extract backend-agnostic plotting data from results
+
+    plot_data, internal_params = _get_plot_data(
+        func=func,
+        params=params,
+        bounds=bounds,
+        func_eval=func_eval,
+        selector=selector,
+        n_gridpoints=n_gridpoints,
+        batch_evaluator=batch_evaluator,
+        n_cores=n_cores,
+    )
+
+    lines_list, marker_list, xlabels, ylabels = _extract_slice_plot_lines_and_labels(
+        plot_data=plot_data,
+        internal_params=internal_params,
+        func_eval=func_eval,
+        param_names=param_names,
+        color=color,
+    )
+
+    # ==================================================================================
+    # Generate the figure
+
+    # Determine y-axis view limits
+    if share_y:
+        lb = plot_data["Function Value"].min()
+        ub = plot_data["Function Value"].max()
+        y_range = ub - lb
+        ub += y_range * expand_yrange
+        lb -= y_range * expand_yrange
+        yrange = (lb, ub)
     else:
-        title_kwargs = None
+        yrange = None
 
+    # Determine x-axis view limits
+    if share_x:
+        lb = plot_data["Parameter Value"].min()
+        ub = plot_data["Parameter Value"].max()
+        xrange = (lb, ub)
+    else:
+        xrange = None
```

Comment:
Can you encapsulate this logic in a function? No need to see this in the main function body IMO.